datafusion/functions/src/string/starts_with.rs

```diff
@@ -98,6 +99,27 @@ impl ScalarUDFImpl for StartsWithFunc {
         }
     }
 
+    fn simplify(
+        &self,
+        args: Vec<Expr>,
+        _info: &dyn SimplifyInfo,
+    ) -> Result<ExprSimplifyResult> {
+        if let Expr::Literal(ScalarValue::Utf8(Some(pattern))) = &args[1] {
+            // Convert starts_with (col, 'prefix') to col LIKE 'prefix%'
+            let like_pattern = format!("{}%", pattern);
```

Comment:
I think we probably need to escape any `%` that appears in the pattern (or avoid doing this replacement when such a pattern exists)